packaging/technical-addon/packaging-scripts/cicd-tests/happypath-test.sh

```diff
@@ -0,0 +1,108 @@
+# TODO use jq not awk
+deployment_id="$(grep "orca_deployment_id" "$TEST_FOLDER/orca_deployment.json" | awk -F ':' '{print $2}' | awk -F '"' '{print $2}')"
+echo "$deployment_id" > "$TEST_FOLDER/deployment_id.txt"
+ip_addr="$(jq -r '.server_roles.standalone[0].host' < "$TEST_FOLDER/orca_deployment.json")"
+
+if [ "$PLATFORM" == "windows" ]; then
+    # Windows takes forever to extract
+    echo "sleeping for 700s at $(date)"
+    sleep 700s
+else
+    # Can likely drop this way down, but give the collector time to collect metrics/traces
+    echo "sleeping for 90s at $(date)"
+    sleep 90s
+fi
+
+
+# Copy logs from container
+scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -r -i ~/.orca/id_rsa "splunk@$ip_addr:/opt/splunk/var/log/splunk/" "$TEST_FOLDER"
+
+# Verify Otel agent is running without any error
+grep -q "Starting otel agent" "$TEST_FOLDER/splunk/Splunk_TA_otel.log"
+grep -q "Everything is ready" "$TEST_FOLDER/splunk/otel.log"
+(grep -qi "ERROR" "$TEST_FOLDER/splunk/Splunk_TA_otel.log" && exit 1 ) || true
+(grep -qi "ERROR" "$TEST_FOLDER/splunk/otel.log" && exit 1 ) || true
+
+# Verify Olly has received metrics data from this host
+MAX_ATTEMPTS=6
+DELAY=10
+ATTEMPT=1
+CUTOFF="$(date '+%s%3N' -d '5 min ago')"
+export CUTOFF
+otel_hostname="$(grep "host.name" "$TEST_FOLDER/splunk/otel.log" | head -1 | awk -F 'host.name":"' '{print $2}' | awk -F '","' '{print $1}')"
+while [ $ATTEMPT -le $MAX_ATTEMPTS ]; do
+    curl --header "Content-Type:application/json" --header "X-SF-TOKEN:${OLLY_ACCESS_TOKEN}" "https://api.us0.signalfx.com/v2/metrictimeseries?query=host.name:${otel_hostname}%20AND%20sf_metric:otelcol_process_uptime%20AND%20splunk.distribution:otel-ta" > "$TEST_FOLDER/uptime.json"
+    count=$( grep '"count"' "$TEST_FOLDER/uptime.json" | awk -F ':\ ' '{print $2}' | awk -F ',' '{print $1}')
+
+    if [[ "$count" -gt "0" ]] && jq '[.results[].created, .results[].lastUpdated] | max as $max | $max >= ($ENV.CUTOFF | tonumber)' "$TEST_FOLDER/uptime.json" ; then
+        break
+    else
+        ATTEMPT=$((ATTEMPT + 1))
+        sleep $DELAY
+    fi
+done
+if [ $ATTEMPT -gt $MAX_ATTEMPTS ]; then
+    echo "Failed find metrics in last 5m after $MAX_ATTEMPTS attempts."
+    cat "$GATEWAY_AGENT_LOGS_DIR/otel.log"
+    exit 1
+fi
+
+# Verify the addon can be restarted successfully
+orca_container_name=$(splunk_orca --cloud ${ORCA_CLOUD} --printer json show --deployment-id "${deployment_id}" containers |  jq -r '.[keys[0]] | .[keys[0]] | .containers | keys[0]')
+splunk_orca --cloud "${ORCA_CLOUD}" exec --exec-user splunk "${orca_container_name}" '/opt/splunk/bin/splunk restart'
+sleep 90s
```

Comment:
Do we need the sleep if we're about to do a loop checking if it's successful with delays in between attempts? Can we just adjust the loop as needed to ensure its check is effective?